### PR TITLE
OCPBUGS-17682: add topologySpreadConstraints to UWM prometheus operator

### PR DESCRIPTION
**Pull request by marioferh**
Add topologySpreadConstraints to UWM prometheus operator
Add test

### CODE REVIEW

File: pkg/manifests/manifests_test.go [hunk 1]
```diff
@@ -4322,9 +4322,19 @@ func TestPodDisruptionBudget(t *testing.T) {
 }
 
 func TestPrometheusOperatorUserWorkloadConfiguration(t *testing.T) {
-	c, err := NewConfigFromString(`
-enableUserWorkload: true
-`, false)
+	c := NewDefaultConfig()
+	uwc, err := NewUserConfigFromString(`prometheusOperator:
+  enableUserWorkload: true
```
Comment (juzhao) [inline pkg/manifests/manifests_test.go:4327]: `enableUserWorkload: true` is not needed, since this func is test UWM prometheus-operator

Comment (marioferh) [reply pkg/manifests/manifests_test.go:4327]: right, thanks

**Pull request merged**